src/pages/PublishedSchedule/PublishedSchedule.jsx

```diff
@@ -52,60 +62,146 @@ const PublishedSchedule = () => {
     setAllSeasons(data);
   };
 
+  const renderTable = async () => {
+    if (selectedSeason.length) {
+      const [seasonType, seasonYear] = selectedSeason.split(' ');
+      const { data } = await NPOBackend.get(
+        `/published-schedule/season?season=${seasonType}&year=${seasonYear}`,
+      );
+      setAllEvents(data);
+      if (!searchedDate.length) {
+        setEventData(data);
+      }
+    }
+  };
+
+  useEffect(() => {
+    renderTable();
+  }, [selectedSeason]);
+
   useEffect(() => {
     if (selectedSeason === '') {
       setSelectedSeason(curSeason);
     }
     renderSeasons();
+    renderTable();
   }, []);
 
   useEffect(() => {
     if (dataShouldRevalidate) {
       renderSeasons();
+      renderTable();
       setShouldDataRevalidate(false);
     }
-  }, [dataShouldRevalidate, renderSeasons])
+  }, [dataShouldRevalidate, renderSeasons, renderTable]);
+
+  useEffect(() => {
+    if (searchedDate) {
+      const searchedData = allEvents.filter(item => item.day.eventDate === searchedDate);
+      setEventData(searchedData);
+    } else {
+      setEventData(allEvents);
+    }
+  }, [searchedDate])
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'allEvents'. Either include it or remove the dependency array. You can also replace multiple useState variables with useReducer if 'setEventData' needs the current value of 'allEvents'.